### PR TITLE
Research into Greening Wagtail

### PR DESCRIPTION
**Pull request by KingNonso**
My base thoughts looking into the - Greener coding: Wagtail’s climate impact

### CODE REVIEW

File: 2023/chinonso-ani/RFC72.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- [Book: Fluent Python, 2nd Edition - Chapter 19. Concurrency Models in Python](https://learning.oreilly.com/library/view/fluent-python-2nd/9781492056348/ch19.html)
+
+
+## Other areas that this can help - Little Language models
+> If content is King, the conversation is Queen - {Attribution incoming}
+Large language models like ChatGPT makes the news, being trained on all or most of the available internet contents, but do we all need Large Language model for all use cases... I think not.
+This is where little language model comes in. Little language model knows a lot about a little.
+
+We already have a CMS, C = Content, so why don't we train a Little Language Model on that content, and possibly build a better chatbot than the rule based system.
+This idea is still in its early stages... To be continued.
```
Comment (thibaudcolas) [inline 2023/chinonso-ani/RFC72.md:79]: :) Knowing the current state of LLMs, we’d have probably tried harder to come up with GSoC ideas around this!

File: 2023/chinonso-ani/Green-Wagtail.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Greener coding: Wagtail’s climate impact
+I read through this project idea, watched @mrchrisadams video from 2020, this is quite a long-tail project requiring substantive research.
+
+## 1. Identify & Understand
+- Understand the climate impact of python/django and by extension wagtail project.
+This involves reading through the green coding project from a couple of sources he listed
```
Comment (thibaudcolas) [inline 2023/chinonso-ani/Green-Wagtail.md:6]: I would also recommend courses like: https://training.linuxfoundation.org/training/green-software-for-practitioners-lfc131/

File: 2023/chinonso-ani/Green-Wagtail.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+* [From Wagtail Discussion: Greener coding - Making a 'gold' reference configuration with the Wagtail bakery app #8843](https://github.com/wagtail/wagtail/discussions/8843)
+* Understanding how [Neon.tech](https://neon.tech/docs/introduction) implemented a serverless PostreSQL - is it possible to implement a serverless wagtail??? (Big weird question - a wagtail that can be deployed to lambda or something similar)
```
Comment (thibaudcolas) [inline 2023/chinonso-ani/Green-Wagtail.md:8]: That would be **amazing** to figure out as part of this internship. The database takes up as much if not more energy than the the server for real-world sites.

File: 2023/chinonso-ani/Green-Wagtail.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+* Code optimization: this is going to be a hard one, but what can be optimized about where/how wagtail runs, is written, is deployed?
```
Comment (thibaudcolas) [inline 2023/chinonso-ani/Green-Wagtail.md:9]: That’s fine to only have a glimpse at. Knowing how to do performance testing of Django / where there might be low-hanging fruits is plenty enough. No need to identify the actual spots at this stage.

File: 2023/chinonso-ani/Green-Wagtail.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## 2. Design
+I think it is important to follow a test driven approach here. To make sure we know when we have done it right. 
+>What does it mean to have done right? (how would done right look like?)
+
+GOLD - Green Open Lean Distributed
+Our design must be golden. (I love this phrase, because I'm the Goldsmith.)
```
Comment (thibaudcolas) [inline 2023/chinonso-ani/Green-Wagtail.md:17]: Don’t get too carried away by the acronym!

File: 2023/chinonso-ani/Green-Wagtail.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Qualitative: 
+2. Quantitative: Ideas discussed [here](https://www.cloudcarbonfootprint.org/docs/methodology/) follow a qualitative measure. Provides a good startup pad.
```
Comment (thibaudcolas) [inline 2023/chinonso-ani/Green-Wagtail.md:20]: 👍 good find

File: 2023/chinonso-ani/RFC72.md (hunk 1 above)
Comment (KingNonso) [reply 2023/chinonso-ani/RFC72.md:79]: I agree to this

File: 2023/chinonso-ani/Green-Wagtail.md (hunk 3 above)
Comment (KingNonso) [reply 2023/chinonso-ani/Green-Wagtail.md:8]: Yeah, listen to the podcast on Neon.tech Postgres db implementation. It's world class! It's on the Changelog Podcast

File: 2023/chinonso-ani/Green-Wagtail.md (hunk 2 above)
Comment (KingNonso) [reply 2023/chinonso-ani/Green-Wagtail.md:6]: Thanks Thibaud, I've signed up. Just the perfect foundation I need! Wow... thanks again.

**Pull request merged**